inference-engine/src/gna_plugin/frontend/layer_quantizer.hpp

```diff
@@ -654,7 +654,28 @@ class DataQuantizer<Desc, InferenceEngine::WeightableLayer *> : public DataQuant
  public:
     explicit DataQuantizer(float scaleFactor) : DataQuantizerBase(scaleFactor) {}
     bool operator()(InferenceEngine::WeightableLayer *wl) const {
-        quantizeWeightsBiases<typename Desc::MandatoryType>(Desc::mandatory(), wl, Quant<typename Desc::MandatoryType>());
+        auto quantData = InferenceEngine::getInjectedData<QuantizedLayerParams>(*wl);
+        if (quantData->_weights_quant.GetLevels() <= 255) {
```

Comment:
for the model without FakeQuantize layers GetLevels() will always return 0 (since they don't have statistics including levels). It seems that with this solution we will always quantize models without FQ to int8. You can use quantData->_weights_quant.isStatsSet() method to know if statistics is available for some layer.

Reply:
done